packages/@apphosting/adapter-nextjs/src/bin/build.ts

```diff
@@ -2,12 +2,58 @@
 import { spawnSync } from "child_process";
 import { loadConfig, readRoutesManifest } from "../utils.js";
 
+import { join } from "path";
+import fsExtra from "fs-extra";
+import { stringify as yamlStringify } from "yaml";
+
+const { move, exists, writeFile, mkdirp } = fsExtra;
 const cwd = process.cwd();
 
-spawnSync("next", ["build"], { cwd, stdio: "inherit", env: { ...process.env, NEXT_PRIVATE_STANDALONE: 'true' } });
+// Set standalone mode
+process.env.NEXT_PRIVATE_STANDALONE = "true"; 
+// Opt-out sending telemetry to Vercel
+process.env.NEXT_TELEMETRY_DISABLED = "1"; 
+
+build(cwd);
+
+const {distDir, basePath} = await loadConfig(cwd);
+const manifest = await readRoutesManifest(join(cwd, distDir));
+
+const destination = join(cwd, ".apphosting");
+const staticDestination = join(destination,"_next", "static");
+const publicDestination = join(destination, "public");
+const outputBundleDestination = join(destination, "bundle.yaml");
+
+const standaloneDir = join (cwd, distDir, "standalone");
+const staticDir = join (cwd, distDir, "static");
+const publicDir = join(cwd, "public");
+
+await mkdirp(staticDestination);
+
+// Run build command
+function build(cwd: string) {
+    spawnSync("npm run", ["build"], {cwd, stdio: "inherit"}); 
```

Comment:
Move run into the second arg, `"npm", ["run", "build"]...`

Also my spidey sense tells me this might need `shell: true` but I'm not sure if that's my being overly defensive, worth investigating e.g, `spawnSync("npm", ["run", "build"], {cwd, shell: true, stdio: "inherit"});`

Reply:
after some research, it seems like having shell: true is indeed very safe, there could possibly be empty output without that due to syntax problems